source/mixin/network_mixin.py

```diff
@@ -0,0 +1,40 @@
+from PyQt4.QtCore import *
+from PyQt4.QtNetwork import *
+
+
+class NetworkMixin(QNetworkAccessManager):
+    request_url = str
+    result_connection = ""
+
+    @pyqtSlot()
+    def connectionFinished(self):
+        print(self.result_connection)
+
+    @pyqtSlot()
+    def connectionReadData(self):
+        self.result_connection += self.reply.readAll()
+
+    def connect(self):
+        print "connect it"
+        QObject.connect(
+            self.reply, SIGNAL('readyRead()'),
+            self, SLOT('connectionReadData()'))
+
+        QObject.connect(
+            self.reply, SIGNAL('finished()'),
+            self, SLOT('connectionFinished()'))
+
+    def connectGET(self):
+        self.reply = self.get(self.req)
+        self.connect()
+
+    def connectPOST(self, data):
+        print "create parameter : %s" % data
```

Comment:
And to add a handler to forward the message to QGIS : https://github.com/inasafe/inasafe/blob/develop/safe/common/custom_logging.py#L43